packages/langium/src/generator/generator-node.ts

```diff
@@ -41,14 +43,29 @@ export function isNewLineNode(node: unknown): node is NewLineNode {
  * @param defaultIndentation the indentation to be applied if no explicit indentation is configured
  *  for particular {@link IndentNode IndentNodes}, either a `string` or a `number` of repeated single spaces,
  *  defaults to 4 single spaces, see {@link processGeneratorNode} -> `Context`.
+ *
+ * @returns the plain `string` represented by the given input.
  */
 export function toString(input: unknown, defaultIndentation?: string | number): string {
     if (isGeneratorNode(input))
-        return processGeneratorNode(input, defaultIndentation);
+        return processGeneratorNode(input, defaultIndentation).text;
     else
         return String(input);
 }
 
+/**
+ * Converts instances of {@link GeneratorNode} into `text` accompanied by a corresponding `trace`.
+ *
+ * @param defaultIndentation the indentation to be applied if no explicit indentation is configured
+ *  for particular {@link IndentNode IndentNodes}, either a `string` or a `number` of repeated single spaces,
+ *  defaults to 4 single spaces, see {@link processGeneratorNode} -> `Context`.
+ *
+ * @returns an object of type `{ text: string, trace: TraceRegion }` containing the desired `text` and `trace` data
+ */
+export function toStringAndTrace(input: GeneratorNode, defaultIndentation?: string | number): { text: string, trace: TraceRegion } {
+    return processGeneratorNode(input, defaultIndentation);
```

Comment:
Why offer the same functionality with two names? Can't we export `processGeneratorNode` to users, or rename it?

Reply:
Maybe yes. I don't want to expose too many parts as API and wanted to leave us freedom of migrating the implementation. For that reason `processGeneratorNode` is not exposed via some `index.ts`.

Reply:
> putting all the stuff in one file makes also little sense

It depends. Microsoft is doing that for parts of the VS Code and TypeScript code bases.